Lib/unittest/mock.py

```diff
@@ -931,13 +931,21 @@ def assert_has_calls(self, calls, any_order=False):
         If `any_order` is True then the calls can be in any order, but
         they must all appear in `mock_calls`."""
         expected = [self._call_matcher(c) for c in calls]
-        cause = expected if isinstance(expected, Exception) else None
+        cause = next((e for e in expected if isinstance(e, Exception)), None)
         all_calls = _CallList(self._call_matcher(c) for c in self.mock_calls)
         if not any_order:
             if expected not in all_calls:
+                if cause is None:
+                    problem = 'Calls not found.'
+                else:
+                    problem = ('Error processing expected calls.\n'
+                               'Errors: {}').format(
+                                   [e if isinstance(e, Exception) else None
+                                    for e in expected])
                 raise AssertionError(
-                    'Calls not found.\nExpected: %r%s'
-                    % (_CallList(calls), self._calls_repr(prefix="Actual"))
+                    f'{problem}\n'
+                    f'Expected: {_CallList(calls)}\n'
+                    f'Actual: {self._calls_repr(prefix="Actual")}'
```

Comment:
This causes a difference in error message with Actual line printed twice. Using `self._calls_repr` takes care of it. `self._calls_repr` already prepends a newline so please remove the "\n" in `f'Expected: {_CallList(calls)}\n'` to avoid an extra new line.

```python
from unittest.mock import *

def f(): pass

m = Mock(spec=f)
m()
m.assert_has_calls([call(), call('wrong')])
```

```
➜  cpython git:(master) ✗ ./python.exe /tmp/foo.py
TypeError: too many positional arguments

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/tmp/foo.py", line 7, in <module>
    m.assert_has_calls([call(), call('wrong')])
  File "/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/mock.py", line 945, in assert_has_calls
    raise AssertionError(
AssertionError: Error processing expected calls.
Errors: [None, TypeError('too many positional arguments')]
Expected: [call(), call('wrong')]
Actual:
Actual: [call()].
```

```suggestion
                    f'{self._calls_repr(prefix="Actual")}'
```

Reply:
whooops. thanks!  i missed that. :)  follow up with a new PR with that little change.  looks like we need to do the backport manually and can lump these two changes together for that anyways.

Reply:
Good catch, sorry. I was making this more similar to cleaner logic in assert_has_awaits, and I missed that repetition.